File: packages/i18nify-js/src/modules/names/__tests__/getHonorificTitles.test.ts
import { getHonorificTitles } from '../index';

describe('names - getHonorificTitles', () => {
  it('returns English honorific titles for US', () => {
    const titles = getHonorificTitles('US');

    expect(titles.length).toBeGreaterThan(0);
    expect(titles).toEqual(
      expect.arrayContaining([
        expect.objectContaining({
          code: 'MR',
          title: 'Mr.',
          gender: 'male',
        }),
      ]),
    );
  });

  it('returns Hindi honorific titles for IN', () => {
    const titles = getHonorificTitles('IN');

    expect(titles.length).toBeGreaterThan(0);
    expect(titles).toEqual(
      expect.arrayContaining([
        expect.objectContaining({
          code: 'SHRI',
          title: 'Shri',
          gender: 'male',
        }),
      ]),
    );
  });

  it('matches country code case-insensitively and trims whitespace', () => {
    expect(getHonorificTitles(' us ')).toEqual(getHonorificTitles('US'));
  });

  it('returns a copy of title entries', () => {
    const titles = getHonorificTitles('US');
    titles[0].title = 'Changed';

    expect(getHonorificTitles('US')[0].title).not.toBe('Changed');
  });

  it('throws for an empty country code', () => {
    expect(() => getHonorificTitles('')).toThrow(/countryCode/);
  });

  it('throws for an unsupported country code', () => {
    expect(() => getHonorificTitles('ZZ')).toThrow(/No honorific titles/);
  });
});

File: packages/i18nify-go/modules/names/get_honorific_titles_test.go
package names

import (
	"testing"

	"github.com/stretchr/testify/assert"
	"github.com/stretchr/testify/require"
)

func TestGetHonorificTitles(t *testing.T) {
	tests := []struct {
		name       string
		country    string
		wantCode   string
		wantTitle  string
		wantGender string
		wantErr    bool
	}{
		{
			name:       "US returns English titles",
			country:    "US",
			wantCode:   "MR",
			wantTitle:  "Mr.",
			wantGender: "male",
		},
		{
			name:       "IN returns Hindi titles",
			country:    "IN",
			wantCode:   "SHRI",
			wantTitle:  "Shri",
			wantGender: "male",
		},
		{
			name:       "country code is trimmed and case-insensitive",
			country:    " us ",
			wantCode:   "MR",
			wantTitle:  "Mr.",
			wantGender: "male",
		},
		{
			name:    "empty country code returns error",
			country: "",
			wantErr: true,
		},
		{
			name:    "unsupported country code returns error",
			country: "ZZ",
			wantErr: true,
		},
	}

	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			titles, err := GetHonorificTitles(tt.country)
			if tt.wantErr {
				assert.Error(t, err)
				return
			}

			require.NoError(t, err)
			require.NotEmpty(t, titles)
			assert.Equal(t, tt.wantCode, titles[0].Code)
			assert.Equal(t, tt.wantTitle, titles[0].Title)
			assert.Equal(t, tt.wantGender, titles[0].Gender)
			assert.NotEmpty(t, titles[0].Description)
		})
	}
}

func TestGetHonorificTitlesReturnsCopy(t *testing.T) {
	titles, err := GetHonorificTitles("US")
	require.NoError(t, err)
	require.NotEmpty(t, titles)

	titles[0].Title = "Changed"

	freshTitles, err := GetHonorificTitles("US")
	require.NoError(t, err)
	require.NotEmpty(t, freshTitles)
	assert.NotEqual(t, "Changed", freshTitles[0].Title)
}
